src/app-modules/docker-compose

```diff
@@ -242,13 +268,23 @@ app_stop() {
     local application_name="$1"
     local manifests_dir="$2"
 
-    $docker_compose_cmd --project-directory "$manifests_dir" down >> "$manifests_dir"/compose.log 2>&1
+    docker_compose_cmd --project-directory "$manifests_dir" down >> "$manifests_dir"/compose.log 2>&1
 }
 
 app_requirements() {
-    echo "$docker_compose_cmd":error
-    echo "$docker_cmd":error
-    echo "$delta_executable":warning
+    local rc=0
+    if ! docker --version > /dev/null < /dev/null 2>&1; then
+        echo "error: cannot find command docker in PATH"
+        echo "error: is Docker installed?"
+        rc=127
+    elif ! docker version > /dev/null; then
+        echo "error: failed to connect with the Docker API"
+        rc=1
+    else
+        _discover_compose
+        rc=$?
+    fi
+    return $rc
```

Comment:
why cant we use the API for required commands?

Reply:
Because validating requirements can be so much more than checking if a command exists in path; as in this example: We first check if docker exist, then check if the docker server is running, then we check if docker compose is installed which is either installed as a plugin or standalone.

Reply:
ok. but we can have both. we do not have to remove all the code around the uniform command-exists checks in the caller of this function, all we need is an extra check on call to `app_requirements` and exit/error handling if it returned non-zero. in this way all the orchestrator-specific logic can be here (as you did) and the common part can be in the caller. whatever you choose, you have to use here `$docker_cmd` and not the plain `docker` command.